src/EditorFeatures/Core/Implementation/ChangeSignature/AbstractChangeSignatureCommandHandler.cs

```diff
@@ -2,24 +2,29 @@
 
 using Microsoft.CodeAnalysis.ChangeSignature;
 using Microsoft.CodeAnalysis.Editor.Host;
-using Microsoft.CodeAnalysis.Editor.Shared;
 using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
 using Microsoft.CodeAnalysis.Editor.Undo;
 using Microsoft.CodeAnalysis.Notification;
 using Microsoft.CodeAnalysis.Shared.Extensions;
-using Microsoft.CodeAnalysis.Text;
 using Microsoft.VisualStudio.Commanding;
 using Microsoft.VisualStudio.Text;
 using Microsoft.VisualStudio.Text.Editor;
 using Microsoft.VisualStudio.Text.Editor.Commanding.Commands;
-using Roslyn.Utilities;
 using VSCommanding = Microsoft.VisualStudio.Commanding;
 
 namespace Microsoft.CodeAnalysis.Editor.Implementation.ChangeSignature
 {
     internal abstract class AbstractChangeSignatureCommandHandler : VSCommanding.ICommandHandler<ReorderParametersCommandArgs>,
         VSCommanding.ICommandHandler<RemoveParametersCommandArgs>
     {
+        private readonly IThreadingContext _threadingContext;
+
+        protected AbstractChangeSignatureCommandHandler(IThreadingContext threadingContext)
+        {
+            _threadingContext = threadingContext;
```

Comment:
I guess most of the updates to pass in the threading context to the constructors should have contracts throw if null?  I'd only push for this one in particular though to establish a pattern in the new constructor.

Reply:
@sharwell @jasonmalinowski actually is threadingcontext exist for all host? like VS for mac, unit test and etc?

Reply:
IThreadingContext exists anywhere the editor does since the editor now depends on JTF, so this code here would be safe to assume it.